### PR TITLE
RC-v1.8 <-- master updates

### PR DESCRIPTION
**Pull request by ap-justin**
Ticket(s):

## Explanation of the solution
* resolve conflicts

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
-

## UI changes for review

When major UI changes are introduced with a PR, please include links to URLS to compare or screenshots demonstrating the difference and notify on design changes

### CODE REVIEW

File: src/pages/Admin/Core/Templates/registrar/Config/useConfigureRegistrar.ts
Comment (misicnenad) [inline src/pages/Admin/Core/Templates/registrar/Config/useConfigureRegistrar.ts:38]: nit: can use `isEmpty(diffEntries)`.

**Review by misicnenad — COMMENTED** [review]
Let's wait for PR https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1860 to be merged to `master`, so those changes are part of `RC-v1.8` as well.

File: src/pages/Admin/Core/Templates/registrar/Config/useConfigureRegistrar.ts
Comment (ap-justin) [reply src/pages/Admin/Core/Templates/registrar/Config/useConfigureRegistrar.ts:38]: these are just cherry picks

**Comment by ap-justin** [comment]
> Let's wait for PR #1860 to be merged to `master`, so those changes are part of `RC-v1.8` as well.

could be cherry-picked later

**Comment by misicnenad** [comment]
> > Let's wait for PR #1860 to be merged to `master`, so those changes are part of `RC-v1.8` as well.
> 
> could be cherry-picked later

@ap-justin the PR in question is merged to `master`, can you merge those new changes and we can merge this PR as well?

**Comment by ap-justin** [comment]
> > > Let's wait for PR #1860 to be merged to `master`, so those changes are part of `RC-v1.8` as well.
> > 
> > 
> > could be cherry-picked later
> 
> @ap-justin the PR in question is merged to `master`, can you merge those new changes and we can merge this PR as well?

could be part of another update, this PR isn't suppose to wait for every open PR to master to be merged, there could be direct changes to RC-v1.8 that could later on have conflict if we don't merge these updates immediately

**Comment by SovereignAndrey** [comment]
> > > > Let's wait for PR #1860 to be merged to `master`, so those changes are part of `RC-v1.8` as well.
> > > 
> > > 
> > > could be cherry-picked later
> > 
> > @ap-justin the PR in question is merged to `master`, can you merge those new changes and we can merge this PR as well?
> 
> could be part of another update, this PR isn't suppose to wait for every open PR to master to be merged, there could be direct changes to RC-v1.8 that could later on have conflict if we don't merge these updates immediately

There is no reason to open yet another PR for moving `v1.7.X` changes into `v1.8` when this one is still open and we have more changes to bring along, since this is just a generic "pull changes upstream" PR.

**Comment by ap-justin** [comment]
> > > > > Let's wait for PR #1860 to be merged to `master`, so those changes are part of `RC-v1.8` as well.
> > > > 
> > > > 
> > > > could be cherry-picked later
> > > 
> > > 
> > > @ap-justin the PR in question is merged to `master`, can you merge those new changes and we can merge this PR as well?
> > 
> > 
> > could be part of another update, this PR isn't suppose to wait for every open PR to master to be merged, there could be direct changes to RC-v1.8 that could later on have conflict if we don't merge these updates immediately
> 
> There is no reason to open yet another PR for moving `v1.7.X` changes into `v1.8` when this one is still open and we have more changes to bring along, since this is just a generic "pull changes upstream" PR.

@SovereignAndrey, @misicnenad,  I was referring to the time when I made this PR, there were only updates say some `A,B,C,D` and some `E,F` still in review. These `A,B,C,D,` could have been merged if not for 

1. Requesting changes on just supposed to be  just `cherry-picked` commits
2. Waiting for `E,F` to be merged

now `E,F` has been merged to master, some `G,H,I` is now open there for review - we could go on this pattern where we keep on waiting for new changes until this PR never gets merged and once changes are made in `RC-v1.8`, I would have to go again and resolve conflicts for `A,B,C,D` which could have prevented if they were merged the moment they were `cherry-picked`

**Comment by SovereignAndrey** [comment]
@ap-justin Please: 
- [ ] update this with the latest master changes
- [ ] address the few merge conflicts

We can get this merged in as soon as that's settled. Thanks! :+1:

**Comment by ap-justin** [comment]
@SovereignAndrey, @misicnenad -  I did try to merge the remaining updates from `master` but unfortunately led to new merge conflicts. Also the last 3 commits can't be cherry picked because of this `fatal/bad object error`. Forcing those updates to this branch via `git merge master` would spawn more conflicts which needlessly need to be resolved again. I hope you would understand that I already spent so much time on this and adding more changes to this PR would potentially cause more error

1. Though I do this with diligence, I can't say that I don't make mistakes when resolving conflicts.
2. And those mistakes have more chance to be spotted if this PR is just covering a week's length of `master` changes. 

I have fixed the merge conflicts from the recent changes in `RC-v1.8` and I hope we could merge these updates now for 
the conflicts resolved here I believe are already verified except for this new one https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1862/commits/cc8e4fef601f7f9a3b273fd428efae57b9757eca where simply choosing `incoming` vs `current` isn't sufficient and a fix need to be done ( choosing `charityReview` instead of `apReview` if not `IS_TEST` )

**Review by SovereignAndrey — APPROVED** [review]
Merging as is to get a clean slate.

**Pull request merged**
